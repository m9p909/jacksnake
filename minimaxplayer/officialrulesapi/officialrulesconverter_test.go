package officialrulesapi_test

import (
	"encoding/json"
	"jacksnake/minimaxplayer/coreplayer"
	. "jacksnake/minimaxplayer/officialrulesapi"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func getGameState1Result() coreplayer.GameBoard {
	food := []coreplayer.Point{
		{X: 3, Y: 1},
		{X: 3, Y: 3},
	}

	hazards := []coreplayer.Point{
		{X: 1, Y: 0},
		{X: 0, Y: 1},
	}

	snakes := []coreplayer.Snake{
		{
			ID:     "1",
			Body:   []coreplayer.Point{{X: 1, Y: 2}, {X: 1, Y: 3}},
			Health: 99,
		},
		{
			ID:     "2",
			Body:   []coreplayer.Point{{X: 3, Y: 2}, {X: 2, Y: 2}, {X: 2, Y: 1}},
			Health: 100,
		},
	}

	return coreplayer.GameBoard{
		Turn:    2,
		Height:  5,
		Width:   4,
		Food:    []coreplayer.Point{},
		Hazards: hazards,
		Snakes:  snakes,
	}
}

func Test_OfficialRulesConverter(t *testing.T) {
	converter := OfficialRulesConverter{}

	board := converter.ConvertToOfficialBoard(getGameState1Result())

	if board.Turn != 2 || board.Height != 5 || board.Width != 4 {
		println("bad board vaue")
		println(json.Marshal(board))
	}

	snaps.MatchSnapshot(t, board.Food)
	snaps.MatchSnapshot(t, board.Hazards)
	snaps.MatchSnapshot(t, board.Snakes)

}
