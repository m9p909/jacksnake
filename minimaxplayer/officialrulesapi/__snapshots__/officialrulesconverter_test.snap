
[Test_OfficialRulesConverter - 1]
[]rules.Point{
    {X:3, Y:1},
    {X:3, Y:3},
}
---

[Test_OfficialRulesConverter - 2]
[]rules.Point{
    {X:1, Y:0},
    {X:0, Y:1},
}
---

[Test_OfficialRulesConverter - 3]
[]rules.Snake{
    {
        ID:   "1",
        Body: {
            {X:1, Y:2},
            {X:1, Y:3},
        },
        Health:           99,
        EliminatedCause:  "",
        EliminatedOnTurn: 0,
        EliminatedBy:     "",
    },
    {
        ID:   "2",
        Body: {
            {X:3, Y:2},
            {X:2, Y:2},
            {X:2, Y:1},
        },
        Health:           100,
        EliminatedCause:  "",
        EliminatedOnTurn: 0,
        EliminatedBy:     "",
    },
}
---

[Test_CoretoRules - 1]
[]rules.Point{
    {X:3, Y:1},
    {X:3, Y:3},
}
---

[Test_CoretoRules - 2]
[]rules.Point{
    {X:1, Y:0},
    {X:0, Y:1},
}
---

[Test_CoretoRules - 3]
[]rules.Snake{
    {
        ID:   "1",
        Body: {
            {X:1, Y:2},
            {X:1, Y:3},
        },
        Health:           99,
        EliminatedCause:  "",
        EliminatedOnTurn: 0,
        EliminatedBy:     "",
    },
    {
        ID:   "2",
        Body: {
            {X:3, Y:2},
            {X:2, Y:2},
            {X:2, Y:1},
        },
        Health:           100,
        EliminatedCause:  "",
        EliminatedOnTurn: 0,
        EliminatedBy:     "",
    },
}
---

[Test_RulestoCore - 1]
[]rules.Point{
    {X:3, Y:1},
    {X:3, Y:3},
}
---

[Test_RulestoCore - 2]
[]rules.Point{
    {X:1, Y:0},
    {X:0, Y:1},
}
---

[Test_RulestoCore - 3]
[]rules.Snake{
    {
        ID:   "1",
        Body: {
            {X:1, Y:2},
            {X:1, Y:3},
        },
        Health:           99,
        EliminatedCause:  "",
        EliminatedOnTurn: 0,
        EliminatedBy:     "",
    },
    {
        ID:   "2",
        Body: {
            {X:3, Y:2},
            {X:2, Y:2},
            {X:2, Y:1},
        },
        Health:           100,
        EliminatedCause:  "",
        EliminatedOnTurn: 0,
        EliminatedBy:     "",
    },
}
---
